### PR TITLE
provide matrix_symbolic_sparse class

### PR DESCRIPTION
**Pull request by dimpase**
sparse matrices over SR are broken without this

This is needed in particular to specialize echelonize(), cf #35653

### CODE REVIEW

File: src/sage/matrix/matrix_symbolic_sparse.pyx [hunk 1]
```diff
@@ -0,0 +1,1034 @@
...
+cdef class Matrix_symbolic_sparse(Matrix_generic_sparse):
+    def echelonize(self, **kwds):
+        """
+        Echelonize using the classical algorithm.
+
+
+        TESTS::
+
+            sage: m = matrix([[cos(pi/5), sin(pi/5)], [-sin(pi/5), cos(pi/5)]])
+            sage: m.echelonize(); m
```
Comment (mkoeppe) [inline src/sage/matrix/matrix_symbolic_sparse.pyx:185]: Is the whole thing copy-pasted from `matrix_symbolic_dense`? 

Note that the copy-pasted doctests from here to the end of the file do not use `sparse=True`, so they won't test your new class.

Comment (dimpase) [reply src/sage/matrix/matrix_symbolic_sparse.pyx:185]: there are indirect doctests with sparse=True elsewhere which test this class

But it looks I forgot a commit somewhere. Thanks for catching this.

Comment (dimpase) [reply src/sage/matrix/matrix_symbolic_sparse.pyx:185]: it's mostly copypasted, of course. It looks as if at some point I just got bored with adding `sparse=True`, and forgot about the rest of the doctests.

Comment (dimpase) [reply src/sage/matrix/matrix_symbolic_sparse.pyx:185]: OK, this is fixed - and sparse symbolic vectors had to be added

**Comment by mkoeppe** [comment]
Likely unrelated, but what's going on with the segfaults in libgap? https://github.com/sagemath/sage/actions/runs/5169419889/jobs/9311663557?pr=35714#step:11:5214

**Review by mkoeppe — APPROVED** [review]
Otherwise LGTM.

**Comment by dimpase** [comment]
> Likely unrelated, but what's going on with the segfaults in libgap?

the previous round of tests didn't show these errors. So it's probably random noise.

**Comment by mkoeppe** [comment]
ok then

**Comment by dimpase** [comment]
in case, I tried to reproduce the libgap segfault using the same random seed, no luck.

**Pull request merged**